### PR TITLE
`target_sda` filters `scenario` to have consistent start year as `ald`

### LINKED ISSUE #346: `target_sda()` outputs `NA`s if `scenario` input contains earlier years than `ald` input
`target_sda()` takes `ald` and `scenario` datasets as input. If `ald` data has a minimum year value that is greater than the minimum year value of `scenario`, the function outputs `NA`s. 

Work-around is to filter the scenario data for `year >= min(ald$year)`.

---

Brief description of the problem

``` r
suppressPackageStartupMessages(library(dplyr))
library(r2dii.analysis)

matched <- tibble::tribble(
  ~id_loan, ~id_direct_loantaker, ~name_direct_loantaker, ~id_intermediate_parent_1, ~name_intermediate_parent_1, ~id_ultimate_parent, ~name_ultimate_parent, ~loan_size_outstanding, ~loan_size_outstanding_currency, ~loan_size_credit_limit, ~loan_size_credit_limit_currency, ~sector_classification_system, ~sector_classification_input_type, ~sector_classification_direct_loantaker, ~fi_type, ~flag_project_finance_loan, ~name_project, ~lei_direct_loantaker, ~isin_direct_loantaker, ~id_2dii,             ~level,  ~sector, ~sector_ald,        ~name,    ~name_ald, ~score,    ~source, ~borderline,
  "L756",                  "7",           "xyz cement",                        NA,                          NA,                 "7",          "xyz cement",                1234566,                           "CAD",                 1234566,                            "CAD",                       "NAICS",                            "Code",                                  327310,   "Loan",                       "No",  "xyz cement",                    NA,                     NA,    "DL3", "direct_loantaker", "cement",    "cement", "xyz cement", "xyz cement",      1, "loanbook",       FALSE,
)

ald <- tibble::tribble(
  ~company_id, ~name_company, ~is_ultimate_owner,  ~sector,           ~technology, ~year, ~plant_location, ~production,   ~production_unit,  ~emission_factor,               ~emission_factor_unit, ~ald_timestamp,
  38998,  "xyz cement",              FALSE, "cement", "integrated facility",  2020,            "AE",      670000, "tonnes of cement",  0.74613363337134, "tonnes of CO2 per tonne of cement",      "2021 Q1",
)

scenario <- tibble::tribble(
  ~scenario_source, ~scenario,  ~sector,  ~region, ~year,  ~emission_factor,               ~emission_factor_unit,
  "etp_2017",    "b2ds", "cement", "global", 2019L, 0.545732650090909, "tonnes of CO2 per tonne of cement",
  "etp_2017",    "b2ds", "cement", "global", 2020L, 0.538350296909091, "tonnes of CO2 per tonne of cement",
)

target_sda(matched, ald, scenario) %>% 
  filter(emission_factor_metric == "target_b2ds")
#> # A tibble: 2 x 4
#>   sector  year emission_factor_metric emission_factor_value
#>   <chr>  <dbl> <chr>                                  <dbl>
#> 1 cement  2019 target_b2ds                               NA
#> 2 cement  2020 target_b2ds                               NA
```

<sup>Created on 2021-07-22 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

### PR DESCRIPTION
**Pull request by jdhoffa**
Closes #346

### CODE REVIEW

File: vignettes/r2dii-analysis.Rmd [hunk 1]
```diff
@@ -104,7 +104,7 @@ sda_targets
 
 ## Visualization
 
-There are a large variety of possible visualizations stemming from the outputs of `target_market_share()` and `target_sda()`. Below, we highlight a couple of common plots that can easily be created using the `r2dii.plot` package. `
+There are a large variety of possible visualizations stemming from the outputs of `target_market_share()` and `target_sda()`. Below, we highlight a couple of common plots that can easily be created using the `r2dii.plot` package.
```
Comment (maurolepore) [inline vignettes/r2dii-analysis.Rmd:107]: It'd be nice to wrap at 80 chars to make diffs easier to see -- line-by-line.

File: R/target_sda.R [hunk 2]
```diff
@@ -263,6 +263,7 @@ compute_ald_adjusted_scenario <- function(data, corporate_economy) {
     ungroup()
 
   data %>%
+    filter(.data$year >= min(corporate_economy$year)) %>%
```
Comment (maurolepore) [inline R/target_sda.R:266]: Just checking, we don't need `na.rm = TRUE` in this call to `min()`?

**Review by maurolepore — APPROVED** [review]
LGTM

**Comment by maurolepore** [comment]
For the record I see one [CI workflow](https://github.com/2DegreesInvesting/r2dii.analysis/pull/347/checks?check_run_id=3135634956) failing, with rdevel:

```
 x Failed to build source package 'stringdist' 
```

This seems unrelated to this PR and I suspect this will fix itself soon.

File: R/target_sda.R (hunk 2 above)
Comment (jdhoffa) [reply R/target_sda.R:266]: I think it wouldn't hurt to add

**Pull request merged**